packages/devtools_app/lib/src/flutter/app.dart

```diff
@@ -173,28 +189,116 @@ class _AlternateCheckedModeBanner extends StatelessWidget {
 class ReportBugAction extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return InkWell(
-      onTap: () async {
-        // TODO(devoncarew): Support analytics.
-        // ga.select(ga.devToolsMain, ga.feedback);
-
-        const reportIssuesUrl = 'https://github.com/flutter/devtools/issues';
-        if (await url_launcher.canLaunch(reportIssuesUrl)) {
-          await url_launcher.launch(reportIssuesUrl);
-        } else {
-          Notifications.of(context).push('Unable to open url.');
-          Notifications.of(context).push('File issues at $reportIssuesUrl.');
-        }
-      },
-      child: Container(
-        width: DevToolsScaffold.actionWidgetSize,
-        height: DevToolsScaffold.actionWidgetSize,
-        alignment: Alignment.center,
-        child: Icon(
-          Icons.bug_report,
-          size: 20.0,
+    return Tooltip(
+      message: 'Report an issue',
+      waitDuration: tooltipWait,
+      child: InkWell(
+        onTap: () async {
+          // TODO(devoncarew): Support analytics.
+          // ga.select(ga.devToolsMain, ga.feedback);
+
+          const reportIssuesUrl = 'https://github.com/flutter/devtools/issues';
+          if (await url_launcher.canLaunch(reportIssuesUrl)) {
+            await url_launcher.launch(reportIssuesUrl);
+          } else {
+            Notifications.of(context).push('Unable to open url.');
+            Notifications.of(context).push('File issues at $reportIssuesUrl.');
+          }
+        },
+        child: Container(
+          width: DevToolsScaffold.actionWidgetSize,
+          height: DevToolsScaffold.actionWidgetSize,
+          alignment: Alignment.center,
+          child: Icon(
+            Icons.bug_report,
+            size: actionsIconSize,
+          ),
         ),
       ),
     );
   }
 }
+
+class OpenSettingsAction extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Tooltip(
+      message: 'Settings',
+      waitDuration: tooltipWait,
+      child: InkWell(
+        onTap: () async {
+          unawaited(showDialog(
+            context: context,
+            builder: (context) => const SettingsDialog(),
+          ));
+        },
+        child: Container(
+          width: DevToolsScaffold.actionWidgetSize,
+          height: DevToolsScaffold.actionWidgetSize,
+          alignment: Alignment.center,
+          child: Icon(
+            Icons.settings,
+            size: actionsIconSize,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+// TODO(devoncarew): Add an analytics setting.
+
+// TODO(devoncarew): Convert the SettingsDialog over to using a controller?
+
+class SettingsDialog extends StatefulWidget {
+  const SettingsDialog({
+    Key key,
+  }) : super(key: key);
+
+  @override
+  _SettingsDialogState createState() => _SettingsDialogState();
+}
+
+class _SettingsDialogState extends State<SettingsDialog> {
```

Comment:
since we aren't keeping any variables / controllers alive in this State, it feels like SettingsDialog should be a stateless widget. 

We should add a provided settings controller to Controllers that Widgets can access via Controllers.of(context). The SettingsController should manage setting states (where setting values are ValueNotifiers) and have methods to modify them. Then widgets like this one can just grab the active SettingsController from Controller.of(context).settings and listening to its notifiers.

This will make it easy for other parts of DevTools to check / listen to active settings.

Reply:
+1 to the controller re-factor. I'd like to do that as a follow-on PR (and added a TODO: for it).

I don't believe that this can be a stateless widget - we do show state in the checkbox below (`value: devtools_theme.isDarkTheme`). If there's a way to refactor to have this stateless, sgtm. I think the code will change in any case once there's a controller and the values are backed by ValueNotifiers.

Reply:
If the checkbox is inside a ValueListenableBuilder, the SettingsDialog widget can be stateless. Either way, can be done as part of a refactor.

Reply:
👍 gotcha; will plan to convert to that once there's a controller